### PR TITLE
compress assets: re-enable, configure

### PR DESCRIPTION
**Pull request by cantsin**
[Link to ZenHub issue.](https://app.zenhub.com/workspaces/doj-crt-intake-scrum-board-5d03bf56c1c8a35d482eca0f/issues/18f/crt-portal/562)

## What does this change?

Compresses JS assets. CSS is already compressed, via gulp and `nanocss`. 

From some rough benchmarks, when viewing a report (`/form/view`), the compressing of javascript itself is negligible (before: 705kb transferred; after: 692kb transferred), but the number of requests goes down from 33 to 22 and so does rendering speed, from ~4.5s to ~2s (when `DOMContentLoaded` triggers).

So overall compressing does offer some benefits. One thing to consider is that bundling (not necessarily minifying) our javascript files would help just as much.

### Author

+ [x] If this is a story, run locally and check to make sure all Acceptance Criteria are met. (If any criteria are unclear, ask about them.)
+ [x] Check for [accessibility](/docs/a11y_plan.md).
+ [x] [Tests pass](https://github.com/USDOJ/crt-portal/#tests).

### Reviewer

+ [ ] If this is a story, run locally and check to make sure all Acceptance Criteria are met. (If any criteria are unclear, ask about them.)
+ [ ] Re-check for [accessibility](/docs/a11y_plan.md).
+ [ ] [Tests pass](https://github.com/USDOJ/crt-portal/#tests).

## Notes for reviewer:

See [PR instructions doc](https://github.com/usdoj/crt-portal/blob/master/docs/pull_requests.md) for full pull request review instructions.

### CODE REVIEW

File: crt_portal/cts_forms/templates/forms/base.html [hunk 1]
```diff
@@ -99,8 +100,10 @@
     </div>
     {% endblock %}
   </body>
+  {% compress js %}
   <script src="{% static 'js/url_params_polyfill.js' %}"></script>
   <script src="{% static 'js/uswds.min.js' %}"></script>
   <script src="{% static 'js/focus_alert.js' %}"></script>
   {% block page_js %}{% endblock %}
```
Comment (cantsin) [inline crt_portal/cts_forms/templates/forms/base.html:107]: Note that most of our per-template JS is in a `page_js` block, so this captures a whole lot of other linked javascript files and inline javascript.

File: crt_portal/crt_portal/settings.py [hunk 2]
```diff
@@ -335,8 +335,7 @@
     ('css', 'compressor_toolkit.precompilers.SCSSCompiler'),
 )
 
-# would like to add this before public release
-COMPRESS_ENABLED = False
+COMPRESS_ENABLED = True
```
Comment (cantsin) [inline crt_portal/crt_portal/settings.py:338]: I considered disabling this for local dev usage, but I like the increased rendering speed. Might cause some issues when debugging, but open to suggestions here.

**Comment by cantsin** [comment]
Now gzipping on s3, hopefully -- but no way to test until we try in dev. (If someone knows how to test otherwise, let me know please)

**Comment by Jkrzy** [comment]
Our usage of `django-compressor` feels wonky, I propose we break this one into two separate efforts/cards:

1. Bundling/Minification of assets
    * I'm not confident the `django-compressor` setup we have is going to play nice with `django-storages` and s3. 
    * Given the small benefit you noticed this feels like a much lower priority to me
2. Serving compressed static files
    * Independent of compressor, `AWS_IS_GZIPPED=True` works fine in my testing w/ `localstack`


Bonus 3: Work towards environment parity and have S3 storage be an option for local development.  https://github.com/localstack/localstack or https://www.npmjs.com/package/s3rver are options. I've got a proof of concept for localstack I can push up.

Feel reasonable @cantsin?

**Comment by cantsin** [comment]
Definitely reasonable. `django-compressor` feels like its not necessary here; maybe we can remove outright, since we already use gulp to copy the js files -- maybe we can go ahead and bundle them up instead there (in `gulpfile.js`). Will do a bit of investigation.

**Comment by Jkrzy** [comment]
I went ahead and cherry-picked out the gzipping into https://github.com/usdoj-crt/crt-portal/pull/750 -- @cantsin, going to close this one.

**Pull request closed without merging**